RSyntaxTextArea/src/main/java/org/fife/ui/rsyntaxtextarea/modes/IniTokenMaker.flex

```diff
@@ -72,151 +72,109 @@ import org.fife.ui.rsyntaxtextarea.*;
+    }
+
+    @Override
+    public String[] getLineCommentStartAndEnd(int languageIndex) {
+        return new String[]{";", null};
+    }
+
+    @Override
+    public Token getTokenList(Segment text, int initialTokenType, int startOffset) {
+        resetTokenList();
+
+        try {
+            super.s = text;
+            super.start = startOffset;
+            reset(text, 0, text.count, 0);
+            yylex();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return new TokenImpl();
+        }
+
+        return firstToken;
+    }
+
+    public final String yytext() {
+        return zzBuffer.subSequence(zzStartRead, zzMarkedPos).toString();
+    }
+
+    @Override
+    public void yyclose() throws IOException {
+    }
+
+    int idStart = -1;
+    int idEnd = -1;
+
+    void updateIdentifier() {
+        if (idStart < 0) {
+            idStart = zzStartRead;
+            idEnd = zzMarkedPos - 1;
+        } else {
+            idEnd = zzMarkedPos - 1;
+        }
+    }
+
+    void addIdentifier(int type) {
+        if (idStart >= 0) {
+            super.addToken(s.array, s.offset + idStart, s.offset + idEnd, type, start + idStart);
+            idStart = idEnd = -1;
+        }
+    }
 %}
 
 Equals				= ([=])
+Escaped             = "\\"[tnr\"f0;#'\\]
 Identifier			= ([^ \t\n#;\[=]*)
 Whitespace			= ([ \t]+)
 Comment				= ([#;].*)
 Section				= ([\[][^\]]*[\]]?)
 
 %state VALUE
+%state STRING
+%state SINGLE_QUOTE_STRING
+%state PRE_VALUE
 
 %%
 
 <YYINITIAL> {
 	{Identifier}		{ addToken(Token.DATA_TYPE); }
-	{Equals}			{ start = zzMarkedPos; addToken(Token.OPERATOR); yybegin(VALUE); }
+	{Equals}			{ addToken(Token.OPERATOR); yybegin(PRE_VALUE); }
 	{Whitespace}		{ addToken(Token.WHITESPACE); }
 	{Comment}			{ addToken(Token.COMMENT_EOL); }
 	{Section}			{ addToken(Token.PREPROCESSOR); }
 	<<EOF>>				{ addNullToken(); return firstToken; }
 }
 
+<PRE_VALUE> {
+    {Whitespace}        { addToken(TokenTypes.WHITESPACE); }
+    [\"]                { addToken(TokenTypes.LITERAL_STRING_DOUBLE_QUOTE); yybegin(STRING); }
+    [']                 { addToken(TokenTypes.LITERAL_STRING_DOUBLE_QUOTE); yybegin(SINGLE_QUOTE_STRING); }
+    [^] 				{ yypushback(1); yybegin(VALUE); }
+    <<EOF>>				{ addNullToken(); return firstToken; }
+}
+
+<STRING> {
+    [\"]                { addIdentifier(TokenTypes.LITERAL_STRING_DOUBLE_QUOTE); addToken(TokenTypes.LITERAL_STRING_DOUBLE_QUOTE); yybegin(VALUE); }
+}
+
+<SINGLE_QUOTE_STRING> {
+    [']                 { addIdentifier(TokenTypes.LITERAL_STRING_DOUBLE_QUOTE); addToken(TokenTypes.LITERAL_STRING_DOUBLE_QUOTE); yybegin(VALUE); }
+}
+
+<STRING, SINGLE_QUOTE_STRING> {
+    {Escaped}           { addIdentifier(TokenTypes.LITERAL_STRING_DOUBLE_QUOTE); addToken(TokenTypes.LITERAL_STRING_DOUBLE_QUOTE); }
+    {Whitespace}        { addIdentifier(TokenTypes.LITERAL_STRING_DOUBLE_QUOTE); addToken(TokenTypes.WHITESPACE); }
+    [^]                 { updateIdentifier(); }
```

Comment:
Same with the above - why are we doing an `addIdentifier` and `addToken` together here? Won't this produce extra tokens?